src/stk/math/types.h

```diff
@@ -131,50 +132,97 @@ struct STK_ALIGN(16) double4
 #endif // STK_USE_CUDA
 
 
-template<typename T, unsigned int rows_, unsigned int cols_>
-struct matrix
+struct Matrix3x3f
 {
-    T _data[rows_][cols_];
+    static constexpr unsigned int rows = 3;
+    static constexpr unsigned int cols = 3;
 
-    static constexpr unsigned int rows = rows_;
-    static constexpr unsigned int cols = cols_;
+    float3 _rows[rows];
```

Comment:
Assuming that we won't do much with this matrix except for left-multiplying it to points, I thought it could be a good idea to just store it as an array of rows.